src/context.rs

```diff
@@ -341,12 +343,115 @@ impl<A: Actor> Context<A> {
         let mut addr_recv = addr_rx.recv_async();
         let mut broadcast_recv = broadcast_rx.recv_async();
 
-        loop {
+        while self.running == RunningState::Running {
+            let (next_msg, unfinished) = {
```

Comment:
This is currently a lot of code duplication, but hopefully if #85 makes it in it should be able to be simplified quite a lot